dav-controller/api/core/acapy/client.py

```diff
@@ -0,0 +1,215 @@
+PUBLIC_WALLET_DID_URI = "/wallet/did/public"
+CREATE_PRESENTATION_REQUEST_URL = "/present-proof/create-request"
+PRESENT_PROOF_RECORDS = "/present-proof/records"
+
+
+class AcapyClient:
+    acapy_host = settings.ACAPY_ADMIN_URL
+    service_endpoint = settings.ACAPY_AGENT_URL
+
+    wallet_token: Optional[str] = None
+    agent_config: AgentConfig
+
+    def __init__(self):
+        if settings.ACAPY_TENANCY == "multi":
+            self.agent_config = MultiTenantAcapy()
+        elif settings.ACAPY_TENANCY == "single":
+            self.agent_config = SingleTenantAcapy()
+        else:
+            logger.warning("ACAPY_TENANCY not set, assuming SingleTenantAcapy")
+            self.agent_config = SingleTenantAcapy()
+
+        if _client:
+            return _client
+        super().__init__()
+
+    def generate_verification_proof_request(
+        self,
+        name: str = "age-verification",
+        nonce: str = "1234567890",
+        age: int = 19,
+        attrib_names_list: List[str] = ["picture"],
+        revocation: bool = True,
+        attrib_ident: str = "picture",
+        predicate_ident: str = "birthdate_GE",
+        cred_def_id: str = None,
+        schema_name: str = None,
+        schema_version: str = None,
+    ):
+        result = {
+            "name": name,
+            "nonce": nonce,
+            "version": "1.0",
+            "requested_attributes": {},
+            "requested_predicates": {},
+        }
+        d = datetime.today()
+        birth_date = datetime(d.year - age, d.month, d.day)
+        birth_date_format = "%Y%m%d"
+        req_attrib_ident = f"0_{attrib_ident}_uuid"
+        req_predicate_ident = f"0_{predicate_ident}_uuid"
+        if schema_name and schema_version:
+            result["requested_attributes"][req_attrib_ident] = {
+                "names": attrib_names_list,
+                "restrictions": [
+                    {
+                        "schema_name": schema_name,
+                        "schema_version": schema_version,
+                    }
+                ],
+            }
+            result["requested_predicates"][req_predicate_ident] = {
+                "name": "birthdate_dateint",
+                "p_type": ">",
+                "p_value": int(birth_date.strftime(birth_date_format)),
+                "restrictions": [
+                    {
+                        "schema_name": schema_name,
+                        "schema_version": schema_version,
+                    }
+                ],
+            }
+        else:
+            result["requested_attributes"][req_attrib_ident] = {
+                "names": attrib_names_list,
+                "restrictions": [
+                    {
+                        "cred_def_id": cred_def_id,
+                    }
+                ],
+            }
+            result["requested_predicates"][req_predicate_ident] = {
+                "name": "birthdate_dateint",
+                "p_type": ">",
```

Comment:
This should be `<=` to check that the birthdate is *before* or equal to the cut-off date.